zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs

```diff
@@ -0,0 +1,132 @@
+        ]
+        .map(|slot| cb.word_rlc(slot.to_le_bytes().map(|b| b.expr())));
+
+        let base_fee = cb.query_word_rlc();
+        let fee_overhead = cb.query_word_rlc();
+        let fee_scalar = cb.query_word_rlc();
+
+        let base_fee_committed = cb.query_cell_phase2();
+        let fee_overhead_committed = cb.query_cell_phase2();
+        let fee_scalar_committed = cb.query_cell_phase2();
+
+        // Read L1 base fee
+        cb.account_storage_read(
+            l1_fee_address.expr(),
+            base_fee_slot,
+            base_fee.expr(),
+            tx_id.expr(),
+            base_fee_committed.expr(),
+        );
+
+        // Read L1 fee overhead
+        cb.account_storage_read(
+            l1_fee_address.expr(),
+            overhead_slot,
+            fee_overhead.expr(),
+            tx_id.expr(),
+            fee_overhead_committed.expr(),
+        );
+
+        // Read L1 fee scalar
+        cb.account_storage_read(
+            l1_fee_address,
+            scalar_slot,
+            fee_scalar.expr(),
+            tx_id,
+            fee_scalar_committed.expr(),
+        );
+
+        Self {
+            base_fee,
+            fee_overhead,
+            fee_scalar,
+            base_fee_committed,
+            fee_overhead_committed,
+            fee_scalar_committed,
+        }
+    }
+
+    pub(crate) fn assign(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        l1_fee: TxL1Fee,
+        l1_fee_committed: TxL1Fee,
+    ) -> Result<(), Error> {
+        self.base_fee
+            .assign(region, offset, Some(l1_fee.base_fee.to_le_bytes()))?;
+        self.fee_overhead
+            .assign(region, offset, Some(l1_fee.fee_overhead.to_le_bytes()))?;
+        self.fee_scalar
+            .assign(region, offset, Some(l1_fee.fee_scalar.to_le_bytes()))?;
+        self.base_fee_committed.assign(
+            region,
+            offset,
+            Value::known(F::from(l1_fee_committed.base_fee)),
+        )?;
+        self.fee_overhead_committed.assign(
+            region,
+            offset,
+            Value::known(F::from(l1_fee_committed.fee_overhead)),
+        )?;
+        self.fee_scalar_committed.assign(
+            region,
+            offset,
+            Value::known(F::from(l1_fee_committed.fee_scalar)),
+        )?;
+
+        Ok(())
+    }
+
+    pub(crate) fn rw_delta(&self) -> Expression<F> {
+        // L1 base fee Read
+        // L1 fee overhead Read
+        // L1 fee scalar Read
+        3.expr()
+    }
+
+    pub(crate) fn tx_l1_fee(&self, tx_call_data_gas_cost: Expression<F>) -> Expression<F> {
+        // <https://github.com/scroll-tech/go-ethereum/blob/49192260a177f1b63fc5ea3b872fb904f396260c/rollup/fees/rollup_fee.go#L118>
+        let tx_l1_gas = tx_call_data_gas_cost + 1088.expr() + self.fee_overhead.expr();
+        self.fee_scalar.expr() * 1_000_000_000.expr() * self.base_fee.expr() * tx_l1_gas
```

Comment:
And here, the `expr()` has given out the rlc value while we need the real value of them. So you need to addup the cells inside `U64Word` by yourself. There is no worry about overflowing of the field since we just mutiple 3x u64 and 1e9.

Reply:
Fix to calculate `tx_l1_fee` with converting words to values by `from_bytes(word.cells[..N_BYTES_U64])` in commit [670207b](https://github.com/scroll-tech/zkevm-circuits/pull/499/commits/670207b0cec0060fb12efdd7b6ec37adeb77ab54).